askama_derive/src/generator.rs

```diff
@@ -1815,7 +1836,21 @@ impl Buffer {
             }
             self.start = false;
         }
-        self.buf.push_str(s);
+
+        if _PREFIX.is_empty() || !s.contains(_ASKAMA_) {
+            self.buf.push_str(s);
```

Comment:
Let's do an early return here, so we can reduce rightward drift in the `else` case?